packages/notifi-react-card/lib/hooks/useSubscriptionCard.ts

```diff
@@ -61,5 +64,14 @@ export const useSubscriptionCard = (
       });
   }, [input.id, input.type]);
 
+  useEffect(() => {
+    if (demoPreview && state.state === 'error') {
+      setState(() => ({
+        state: 'fetched',
+        data: demoPreview.data,
+      }));
+    }
```

Comment:
let's move this into the above useEffect, before doing anything else

Reply:
IMO, It might not be possible to move it to the above useEffect since the 'client.fetchSubscriptionCard' is an async func that makes the useEffect to loop infinitely if we add the  `state` as dependency (state.state change between loading and fetched) .

Reply:
Combined to only one useEffect.